osu.Game/Graphics/Containers/SectionsContainer.cs

```diff
@@ -150,31 +158,63 @@ public override void Add(T drawable)
             footerHeight = null;
         }
 
+        private ScheduledDelegate? scrollToTargetDelegate;
+
         public void ScrollTo(Drawable target)
         {
-            lastKnownScroll = null;
+            Logger.Log($"Scrolling to {target}..");
 
-            // implementation similar to ScrollIntoView but a bit more nuanced.
-            float top = scrollContainer.GetChildPosInContent(target);
+            lastKnownScroll = null;
 
-            float bottomScrollExtent = scrollContainer.ScrollableExtent;
-            float scrollTarget = top - scrollContainer.DisplayableContent * scroll_y_centre;
+            float scrollTarget = getScrollTargetForDrawable(target);
 
-            if (scrollTarget > bottomScrollExtent)
+            if (scrollTarget > scrollContainer.ScrollableExtent)
                 scrollContainer.ScrollToEnd();
             else
                 scrollContainer.ScrollTo(scrollTarget);
 
             if (target is T section)
                 lastClickedSection = section;
+
+            // Content may load in as a scroll occurs, changing the scroll target we need to aim for.
+            // This scheduled operation ensures that we keep trying until actually arriving at the target.
+            scrollToTargetDelegate?.Cancel();
+            scrollToTargetDelegate = Scheduler.AddDelayed(() =>
+            {
+                if (scrollContainer.UserScrolling)
```

Comment:
There's a very small chance this could become `true` and then subsequently `false` before the next schedule of this code is run (if the user scrolls but then performs a code-run scroll operation). The chances of this are very slim, but if it's seen as an issue, either moving this whole block to `UpdateAfterChildren` (or just this `if` portion) would fix that.

Reply:
I say we ignore it until anyone actually hits that.